atc/engine/builder/delegate_factory.go

```diff
@@ -277,25 +280,46 @@ func (d *taskDelegate) Finished(logger lager.Logger, exitStatus exec.ExitStatus)
 func NewBuildStepDelegate(
 	build db.Build,
 	planID atc.PlanID,
+	credVarsTracker vars.CredVarsTracker,
 	clock clock.Clock,
 ) *buildStepDelegate {
 	return &buildStepDelegate{
-		build:  build,
-		planID: planID,
-		clock:  clock,
+		build:           build,
+		planID:          planID,
+		clock:           clock,
+		credVarsTracker: credVarsTracker,
 	}
 }
 
 type buildStepDelegate struct {
-	build  db.Build
-	planID atc.PlanID
-	clock  clock.Clock
+	build           db.Build
+	planID          atc.PlanID
+	clock           clock.Clock
+	credVarsTracker vars.CredVarsTracker
+}
+
+func (delegate *buildStepDelegate) GetCredVarsTracker() vars.CredVarsTracker {
+	return delegate.credVarsTracker
 }
 
 func (delegate *buildStepDelegate) ImageVersionDetermined(resourceCache db.UsedResourceCache) error {
 	return delegate.build.SaveImageResourceVersion(resourceCache)
 }
 
+type credVarsIterator struct {
+	line string
+}
+
+func (it *credVarsIterator) YieldCred(k, v string) {
+	it.line = strings.Replace(it.line, v, fmt.Sprintf("[redacted **%s**]", k), -1)
```

Comment:
Do you think there's any security concern with printing the var that was redacted here? :thinking: 

I can't think of a great example, but if someone were to be able to deduce (from context) the value that was redacted - e.g. a username - then the var may reveal what service the username is for. Granted, they'd still need the password, but I suspect some folks would be uncomfortable with any amount of information leakage.

If there's no real reason to print the var it might be safer to omit it.

Reply:
I don't intend to leave var in redacted text. As this is just the initial change, I printed var in order to make debugging easier. I'll remove var in the final commit.